src/broadcast/scrobblingmanager.cpp

```diff
@@ -106,7 +106,8 @@ void ScrobblingManager::slotTrackPaused(TrackPointer pPausedTrack) {
     TrackInfo *pausedTrackInfo = nullptr;
     for (TrackInfo *trackInfo : m_trackList) {
         DEBUG_ASSERT(trackInfo);
-        if (trackInfo->m_pTrack == pPausedTrack) {
+        if (!trackInfo->m_pTrack.expired() &&
```

Comment:
The track pointer can expire just after the check. Use lock() and check for nullptr afterwards.

Reply:
That was my worry, makes sense.